### PR TITLE
Replace crlsets with redirector

### PR DESCRIPTION
**Pull request by linhkikuchi**
Resolves https://github.com/brave/brave-browser/issues/20609
For https://github.com/brave/devops/issues/3313 

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

Use Proxyman (MacOS) or mitmproxy, run the browser with fresh profile using the proxy ( e.g. `--proxy-server="localhost:9090"`), verify:
- spellcheck dict request is 200 `https://dict.brave.com/edgedl/chrome/dict/en-gb-9-0.bdic` (the old request was `https://crlsets.brave.com/edgedl/chrome/dict/en-gb-9-0.bdic`)
- all other `/edgedl/release2/chrome_component/`, `/release2/chrome_component/`, `/dl/release2/chrome_component`, `/update-delta/hfnkpimlhhgieaddgfemjhofmfblmnib/`, `/edgedl/chromewebstore/`, `/edgedl/release2/`, `/update-delta/ggkkehgbnfjpeggfpleeakpidbkibbmn/` requests are now using `https://redirector.brave.com` and all return 200
- other componentupdate requests should return 200 as well
- go to brave://components and able to update `CRLSet`
Screen Shot 2022-01-21 at 11 22 17

### CODE REVIEW

File: browser/net/brave_static_redirect_network_delegate_helper_unittest.cc [hunk 1]
```diff
@@ -75,7 +75,7 @@ TEST(BraveStaticRedirectNetworkDelegateHelperTest, ModifyCRLSet3) {
       "https://www.google.com/dl/release2/chrome_component/LLjIBPPmveI_4988/"
       "4988_all_crl-set-6296993568184466307.data.crx3");
   const GURL expected_url(
-      "https://crlsets.brave.com/dl/release2/chrome_component/LLjIBPPmveI_4988/"
+      "https://redirector.brave.com/dl/release2/chrome_component/LLjIBPPmveI_4988/"
```
Comment (goodov) [inline browser/net/brave_static_redirect_network_delegate_helper_unittest.cc:78]: some of these longs lines should be clang-formatted

**Comment by iefremov** [comment]
Will need to update 
https://github.com/brave/devops/wiki/Mappings-for-proxied-URLs
https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)#services-we-proxy-through-brave-servers

**Review by iefremov — APPROVED** [review]

File: browser/net/brave_static_redirect_network_delegate_helper_unittest.cc (hunk 1 above)
Comment (linhkikuchi) [reply browser/net/brave_static_redirect_network_delegate_helper_unittest.cc]: thank you, have updated

**Review by goodov — APPROVED** [review]

**Review by fmarier — APPROVED** [review]

**Comment by linhkikuchi** [comment]
> Will need to update https://github.com/brave/devops/wiki/Mappings-for-proxied-URLs https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)#services-we-proxy-through-brave-servers

Updated

**Pull request merged**